source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```
@@ -533,12 +491,134 @@ namespace AdaptiveCards::Rendering::Uwp
+                        {
+                            auto bytes = result.GetResults();
+                            auto svgText = streamDataReader.ReadString(bytes);
+
+                            auto svgDocument = winrt::XmlDocument();
+                            svgDocument.LoadXml(svgText);
+
+                            strongThis->ParseXmlForHeightAndWidth(svgDocument, imageSource);
+                        }
+
+                        // Now that we've parsed the height, we can set the image source
+                        strongThis->SetSvgImageSourceAsync(
+                            imageSource.as<winrt::SvgImageSource>(), stream, uiElement, isAutoSize, parentElement, imageContainer, isVisible);
+                    }
+                });
+        }
+        else
+        {
+            auto setSourceAction = imageSource.as<winrt::BitmapImage>().SetSourceAsync(stream);
+
+            setSourceAction.Completed([weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                winrt::IAsyncAction const& /*operation*/, winrt::AsyncStatus status)
+                {
+                    if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                    {
+                        if (auto strongThis = weakThis.get())
+                        {
+                            strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                        }
+                    }
+                });
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::SvgImageSource image, winrt::Uri imageUrl)
+    {
+        try
+        {
+            co_await winrt::resume_foreground(image.Dispatcher());
+            image.as<winrt::SvgImageSource>().UriSource(imageUrl);
+        }
+        catch (winrt::hresult_wrong_thread ex)
+        {
+            // We didn't switch to the UI thread
```

I don't think we need to fail fast here necessarily but logging the error would be nice :)

---

Added an error statement and passed the `renderContext` to these methods